xarray/tests/test_backends_api.py

```diff
@@ -48,6 +49,32 @@ def open_dataset(
     assert_identical(expected, actual)
 
 
+def test_multiindex() -> None:
+    # GH7139
+    # Check that we properly handle backends that change index variables
+    def _create_multiindex(**kwargs):
+        return pd.MultiIndex.from_arrays(list(kwargs.values()), names=kwargs.keys())
+
+    dataset = xr.Dataset()
+    dataset.coords["coord1"] = ["A", "B"]
+    dataset.coords["coord2"] = [1, 2]
+    dataset["multi_index"] = _create_multiindex(
+        coord1=["A", "A", "B", "B"], coord2=[1, 2, 1, 2]
+    )
```

Comment:
```suggestion
    dataset = xr.Dataset(coords={"coord1": ["A", "B"], "coord2": [1, 2]})
    dataset = dataset.stack(z=["coord1", "coord2"])
```

Directly assigning a pandas multi-index to a dataset (with automatic creation of level coordinates) will eventually be deprecated.

Reply:
Thanks + much more readable way of creating such a dataset, I changed the test now 👍